File: Algorithm/binary_tree.py
class BinaryTree:
    def __init__(self,key):
        self.key=key
        self.left=None
        self.right=None

    #Traverse Inorder
    def traverse_inorder(self):
        if self.left:
            self.left.traverse_inorder()
        print(self.key,end= ' ')
        if self.right:
            self.right.traverse_inorder()

    #Traverse postorder
    def traverse_postorder(self):
        if self.left:
            self.left.traverse_postorder()
        if self.right:
            self.right.traverse_postorder()
        print(self.key, end=' ')

    #Traverse Preorder
    def traverse_preorder(self):
        print(self.key, end=' ')
        if self.left:
            self.left.traverse_preorder()
        if self.right:
            self.right.traverse_preorder()
    #Insert function
    def insert(self, value):
        if value < self.key:
            if self.left is None:
                self.left = BinaryTree(value)
            else:
                self.left.insert(value)
        else:
            if self.right is None:
                self.right = BinaryTree(value)
            else:
                self.right.insert(value)

# Create a binary tree and insert values
root = BinaryTree(5)
root.insert(3)
root.insert(8)
root.insert(2)
root.insert(4)
root.insert(7)
root.insert(9)

# Traverse and print the tree in different orders
print("Inorder Traversal: ", end="")
root.traverse_inorder()
print("\nPostorder Traversal: ", end="")
root.traverse_postorder()
print("\nPreorder Traversal: ", end="")
root.traverse_preorder()

File: binary_tree.java
class Node {
    int data;
    Node left, right;

    public Node(int data) {
        this.data=data;
        left = right = null;
    }
}

class BinaryTree {
    Node root;

    //preorder
    void preorder(Node node) {
        if (node == null)
            return;
        System.out.print(node.data + " ");
        preorder(node.left);
        preorder(node.right);
    }

    //InOrder
    void inorder(Node node1){
        if (node1==null){
            return;
        }
        inorder(node1.left);
        System.out.print(node1.data + " ");
        inorder(node1.right);
    }

    //PostOrder
    void postorder(Node node2){
        if(node2==null){
            return ;
        }
        postorder(node2.left);
        postorder(node2.right);
        System.out.println(node2.data+ " ");
    }

    public static void main(String[] args) {
        BinaryTree tree = new BinaryTree();
        tree.root = new Node(1);
        tree.root.left = new Node(2);
        tree.root.right = new Node(3);
        tree.root.left.left = new Node(4);
        tree.root.left.right = new Node(5);
        tree.root.right.left = new Node(6);
        tree.root.right.right = new Node(7);

        tree.preorder(tree.root);
        System.out.println("\n");
        tree.inorder(tree.root);
        System.out.println("\n");
        tree.postorder(tree.root);
    }
}
